ops/charm.py

```diff
@@ -795,16 +795,35 @@ class RelationMeta:
         role: This is :class:`RelationRole`; one of peer/requires/provides
         relation_name: Name of this relation from metadata.yaml
         interface_name: Optional definition of the interface protocol.
-        scope: "global" or "container" scope based on how the relation should be used.
+        limit: Optional definition of maximum number of connections to this relation endpoint.
+        scope: "global" (default) or "container" scope based on how the relation should be used.
     """
 
+    VALID_SCOPES = ['global', 'container']
+    DEFAULT_SCOPE = VALID_SCOPES[0]
+
     def __init__(self, role: RelationRole, relation_name: str, raw: dict):
         if not isinstance(role, RelationRole):
             raise TypeError("role should be a Role, not {!r}".format(role))
         self.role = role
         self.relation_name = relation_name
         self.interface_name = raw['interface']
-        self.scope = raw.get('scope')
+
+        limit = raw.get('limit')
+        if limit is not None:
+            if type(limit) is not int:
+                raise TypeError("limit should be an int, not {}".format(type(limit)))
+
+            self.limit = int(limit)
+        else:
+            self.limit = None
+
+        scope = raw.get('scope')
+        if scope is not None and scope not in RelationMeta.VALID_SCOPES:
+            valid_scopes = "'" + "', '".join(RelationMeta.VALID_SCOPES) + "'"
+            raise TypeError("scope should be one of {}; not '{}'".format(valid_scopes, scope))
```

Comment:
This is confusing to read:

    raise TypeError("scope should be one of '{}'; not '{}'".format(
                    ', '.join(VALID_SCOPES), scope))